cli/src/anipy_cli/menus/seasonal_menu.py

```diff
@@ -180,65 +186,37 @@ def change_lang(self):
                 print(f"> {new_lang} is for {e.name} not available")
 
     def list_animes(self):
-        for i in self.seasonal_list.get_all():
+        all_seasonals = self.seasonal_list.get_all()
+        if not all_seasonals:
+            error("No seasonals configured.")
+            return
+
+        for i in all_seasonals:
             print(i)
 
     def download_latest(self):
         picked = self._choose_latest()
-        config = Config()
         total_eps = sum([len(e) for _a, _d, e in picked])
         if total_eps == 0:
-            print("Nothing to download, returning...")
+            print("All up to date on downloads.")
             return
         else:
             print(f"Downloading a total of {total_eps} episode(s)")
-        with DotSpinner("Starting Download...") as s:
-
-            def progress_indicator(percentage: float):
-                s.set_text(f"Progress: {percentage:.1f}%")
-
-            def info_display(message: str):
-                s.write(f"> {message}")
-
-            downloader = Downloader(progress_indicator, info_display)
-
-            for anime, lang, eps in picked:
-                for ep in eps:
-                    s.set_text(
-                        "Extracting streams for ",
-                        colors.BLUE,
-                        f"{anime.name} ({lang})",
-                        colors.END,
-                        " Episode ",
-                        ep,
-                        "...",
-                    )
 
-                    stream = anime.get_video(
-                        ep, lang, preferred_quality=self.options.quality
-                    )
+        def onSuccessfulDownload(anime: Anime, ep: Episode, lang: LanguageTypeEnum):
+            self.seasonal_list.update(anime, episode=ep, language=lang)
 
-                    info_display(
-                        f"Downloading Episode {stream.episode} of {anime.name}"
-                    )
-                    s.set_text("Downloading...")
-
-                    downloader.download(
-                        stream,
-                        get_download_path(anime, stream, parent_directory=self.dl_path),
-                        container=config.remux_to,
-                        ffmpeg=self.options.ffmpeg or config.ffmpeg_hls,
-                    )
-                    self.seasonal_list.update(anime, episode=ep, language=lang)
+        failed_series = DownloadComponent(self.options, self.dl_path).download_anime(picked, onSuccessfulDownload)
 
         if not self.options.auto_update:
-            self.print_options(clear_screen=True)
+            # Clear screen only if there were no issues
+            self.print_options(clear_screen=len(failed_series)==0)
```

Comment:
nit: proper spacing here
```suggestion
            self.print_options(clear_screen=len(failed_series) == 0)
```